src/concurrency.md

```diff
@@ -30,6 +31,36 @@ The example uses the Rayon crate, which is a data parallelism library for Rust.
 Rayon provides the `par_iter_mut()` method for any parallel iterable data type.
 It lets us write iterator-like chains that execute in parallel.
 
+[ex-rayon-parallel-sort]: #ex-rayon-parallel-sort
+<a name="ex-rayon-parallel-sort"></a>
+## Sort a vector in parallel
+
+[![rayon-badge]][rayon] [![rand-badge]][rand] [![cat-concurrency-badge]][cat-concurrency]
+
+```rust
+extern crate rand;
+extern crate rayon;
+
+use rand::Rng;
+use rayon::prelude::*;
+
+fn main() {
+    let mut vec = vec![String::new(); 100_000];
+
+    // Generating random 5 character-long Strings to be sorted later
+    vec.par_iter_mut().for_each(|p| {
+        *p = rand::weak_rng().gen_ascii_chars().take(5).collect()
+    });
+
+    // Using Rayon's parallel sort
+    vec.par_sort_unstable(); // unstable sort, like QuickSort
+}
+```
+
+The example uses the `rayon` crate, which is a data parallelism library for Rust.<br/>
+We have [`multiple options`] to sort an Iterable data type, we chose here to use [`par_sort_unstable`]
```

Comment:
Actually I am quite aware of Qt docs as I've used it more than a few times and I can give a even better example of developer docs as Imho Apple developer docs are the gold standard on terms of quality and completeness (hmm somehow the Qt4 docs always seamed nicer than the Qt5 ones. Probably Nokia had more focus on quality then but I'm beside the point :smile_cat:).

But the real point being is that the cookbook is not an API/language nor crate doc (and your points are excellent for such types of resources ). Cookbook is an introductory (and in the future hopefully intermediate) resource assisting problem solving and crate discoverability (and lastly some good pointers on ~~pythonic~~ rustic code).

The typical use case for our reader is intended to:
1. quickly search for solution to a given problem - we are limping here ATM 
2. see what are the actors / crates for typical idiomatic solution - with links to the real docs allowing reader to educate oneself
3. copy the minimal snippet to their code base 

In such light the possible educational value is assumed to come from the linked docs.

Please note that, this is the current working idea that we are following but we are open for suggestions. And we would love to hear yours and other developer's thoughts on this matter. But any change to the cookbook formula will have to be discussed first and then implemented consistently throughout the whole cookbook.

So to summarize. We plan on tweaking the cookbook formula in the future (both content and form) and would love for you to take part in the discussion.